JiraReporter/IndividualReportInfoService.cs

```diff
@@ -0,0 +1,69 @@
+﻿using JiraReporter.Model;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace JiraReporter
+{
+    class IndividualReportInfoService
+    {
+        public void SetIndividualDraftInfo(List<Author> authors, SourceControlLogReporter.Model.Policy policy)
+        {
+            if(!policy.AdvancedOptions.NoIndividualDraft && policy.GeneratedProperties.IsIndividualDraft)
+            {
+                var individualDrafts = new List<SourceControlLogReporter.Model.IndividualDraftInfo>();
+                var random = new Random();
+
+                foreach(var author in authors)
+                {
+                    var individualDraft = GenerateIndividualDraftInfo(author, policy, random);
+                    individualDrafts.Add(individualDraft);
+                    author.IndividualDraftInfo = individualDraft;
+                }
+
+                policy.GeneratedProperties.IndividualDrafts = individualDrafts;
+            }
+        }
+
+        private SourceControlLogReporter.Model.IndividualDraftInfo GenerateIndividualDraftInfo(Author author, SourceControlLogReporter.Model.Policy policy, Random random)
+        {
+            var individualDraft = new SourceControlLogReporter.Model.IndividualDraftInfo
+            {
+                Name = author.Name,
+                Username = author.Username,
+                UserKey = RandomGenerator.RandomString(10, random),
+                IsLead = author.IsProjectLead
+            };
+            SetIndividualUrls(individualDraft, policy);
+
+            return individualDraft;
+        }
+
+        private void SetIndividualUrls(SourceControlLogReporter.Model.IndividualDraftInfo individualDraft, SourceControlLogReporter.Model.Policy policy)
+        {
+            individualDraft.ConfirmationDraftUrl = GetUrl(individualDraft, policy.IndividualDraftConfirmationUrl);
```

Comment:
Url-urile astea nu cred ca e nevoie sa le calculezi in avans si sa le pui pe obiectul asta. Ele ar putea fi f bine calculate exact cand ai nevoie de ele pe baza userkey-ului, nu?